src/tests/ftest/security/pool_security_acl.py

```diff
@@ -0,0 +1,403 @@
+
+def add_del_user(hosts, ba_cmd, user):
+    '''
+    Deascription:
+        Add or delete the daos user and group on host by sudo command.
+    Args:
+        hosts: list of host.
+        ba_cmd: linux bash command to create user or group.
+        user: user or group name to be created or cleaned.
+    Return:
+        none.
+    '''
+    bash_cmd = os.path.join("/usr/sbin", ba_cmd)
+    homedir = ""
+    if "user" in ba_cmd:
+        homedir = "-r"
+    cmd = " ".join(("sudo", bash_cmd, homedir, user))
+    pcmd(hosts, cmd, False)
+
+class PoolSecurityTest(TestWithServers):
+    '''
+    Test Class Description:
+    Tests to verify the Pool security with dmg and daos tools on daos_server.
+    :avocado: recursive
+    '''
+
+    def get_pool_acl_list(self, uuid):
+        '''
+        Deascription:
+            Get daos pool acl list by dmg get-acl
+        Args:
+            uuid: pool uuid number.
+        Return:
+            pool_permission_list: daos pool acl list.
+        '''
+        dmg = DmgCommand(os.path.join(self.prefix, "bin"))
+        dmg.request.value = "pool"
+        dmg.action.value = "get-acl --pool " + uuid
+        port = self.params.get("port", "/run/server_config/*")
+        servers_with_ports = [
+            "{}:{}".format(host, port) for host in self.hostlist_servers]
+        dmg.hostlist.update(",".join(servers_with_ports), "dmg.hostlist")
+        result = dmg.run()
+
+        pool_permission_list = []
+        for line in result.stdout.splitlines():
+            if not line.startswith("A:"):
+                continue
+            elif line.startswith("A::"):
+                found_user = re.search(r"A::(.+)@:(.*)", line)
+                if found_user:
+                    username = found_user.group(1)
+                    userpermission = found_user.group(2)
+                    pool_permission_list.append(line)
+            elif line.startswith("A:G:"):
+                found_group = re.search(r"A:G:(.+)@:(.*)", line)
+                if found_group:
+                    groupname = found_group.group(1)
+                    grouppermission = found_group.group(2)
+                    pool_permission_list.append(line)
+        return pool_permission_list
+
+    def verify_daos_pool_result(self, result, action, expect, err_code):
+        '''
+        Deascription:
+            To verify the daos pool read or write action result.
+        Args:
+            result: handle for daos pool action.
+            action: daos pool read or write.
+            expect: expecting pass or deny.
+            err_code: expecting deny of RC code.
+        Return:
+            none.
+        '''
+        if expect.lower() == 'pass':
+            if result.exit_status != 0 or result.stderr != "":
+                self.fail("##Test Fail on verify_daos_pool {}, "
+                          "expected Pass, but Failed.".format(action))
+            else:
+                self.log.info(" =Test Passed on verify_daos_pool {}, "
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments